clang/lib/Basic/Targets/EVM.cpp

```diff
@@ -25,6 +25,21 @@ const Builtin::Info EVMTargetInfo::BuiltinInfo[] = {
 #include "clang/Basic/BuiltinsEVM.def"
 };
 
+EVMTargetInfo::EVMTargetInfo(const llvm::Triple &Triple, const TargetOptions &)
+    : TargetInfo(Triple) {
+  PointerWidth = 64;
+  BoolWidth = 8;
+  CharWidth = 8;
+  IntWidth = 32;
+  LongWidth = 64;
+  LongLongWidth = 64;
+  MaxBitIntWidth = 64;
+  LongLongAlign = 32;
+  Int256Align = 32;
+  resetDataLayout("E-S64-a:0:32-p:64:64-i1:8:8-i8:8:8-i16:16:16-i32"
```

Comment:
This layout specifies how data is to be laid out in memory. I changed alignment of 128- and 256-bit integers because there was unnecessary overhead and some bugs with it.

Reply:
It specifies how primitive types should be laid in the memory, i.e. what endianness should it use, what is the alignment of pointer and other primitive types. You can read more about it here https://llvm.org/docs/LangRef.html#data-layout
Without this change, the compiler tried to place uint256 with 256-bits alignment and performs truncation of the pointer, that, for some reason, worked incorrectly in some cases.